src/Mod/Draft/draftfunctions/move.py

```diff
@@ -215,4 +218,208 @@ def copy_moved_edge(object, edge_index, vector):
+            new_shape = shape
+
+    return new_shape, touched
+
+
+def sort_wires(wires):
+    if not isinstance(wires, list):
+        return wires
+    if len(wires) == 1:
+        return wires
+    outer_wire = wires[0]
+
+    for w in wires:
+        if outer_wire.BoundBox.DiagonalLength < w.BoundBox.DiagonalLength:
+            outer_wire = w
+    
+    new_wires = [outer_wire]
+    for w in wires:
+        if w != outer_wire:
+            new_wires.append(w)
+    
+    return new_wires
+
+    
+def parse_sub_shapes(shape, selected_subelements, vector):
+    """ Parse the subshapes of the given shape in order to
+    find modified shapes and substitute them to the originals.
+    """
+    sub_shapes = []
+    touched_subshapes = []
+    if shape.SubShapes:
+        for sub_shape in shape.SubShapes:
+            new_sub_shape, touched_subshape = parse_shape(sub_shape, selected_subelements, vector)
+            sub_shapes.append(new_sub_shape)
+
+            if touched_subshape:
+                touched_subshapes.append(2)
+            else:
+                touched_subshapes.append(0)
+
+    if 0 in touched_subshapes and 2 in touched_subshapes:
+        # only some subshapes touched
+        touched = 1
+    elif 2 in touched_subshapes:
+        # all subshapes touched
+        touched = 2
+    elif 0 in touched_subshapes:
+        # no subshapes touched
+        touched = 0
+
+    return sub_shapes, touched
+
+
+def moveSubElements(obj, sub_objects_names, vector):
+    """moveSubElements(obj, sub_objects_names, vector)
+    
+    Move the given object sub_objects according to a vector or crates an new one
+    if the object is not a Part::Feature.
+
+    Parameters
+    ----------
+    obj : the given object
+
+    sub_objects_names : list of subelement names
+        A list of subelement names to identify the subelements to move.
+
+    vector : Base.Vector
+        Delta Vector to move subobjects from the original position. 
+
+    Return
+    ----------
+    shape : Part.Shape
+        Return the new Shape or None.
+    """
+
+    import Part
+
+    if not isinstance(sub_objects_names, list):
+        sub_objects_names = [sub_objects_names]
+
+    shape = obj.Shape
+    new_shape = None
+    if not shape.isValid():
+        return
+
+    selected_subelements = []
+    for sub_objects_name in sub_objects_names:
+        sub_object = obj.Shape.getElement(sub_objects_name)
+        selected_subelements.append(sub_object)
+        selected_subelements.extend(sub_object.Faces)
+        selected_subelements.extend(sub_object.Edges)
+        selected_subelements.extend(sub_object.Vertexes)
+
+    new_shape, touched = parse_shape(shape, selected_subelements, vector)
+    
+    if not new_shape.isValid():
+        should_fix = move_subelements_msgbox()
+        if should_fix:
+            new_shape.fix(0.001,0.001,0.001)
+
+    if new_shape:
+        if hasattr(obj, 'TypeId') and obj.TypeId == 'Part::Feature':
+            obj.Shape = new_shape
```

Comment:
here I wanted explicitly to test for Part::Feature, because if the object is not, we will create a new one, else we just substitute it's shape (cause it's not parametric).
Regarding App::Link, it's probably just a matter of editing the object as we are doing, but checking before (in the gui) which object we are editing and if we have to correct the vector to match the exact placement of the Link object...